src/modules/services_manager_service.rb

```diff
@@ -127,28 +169,35 @@ def extract_services
         # Add old LSB services (Services which are loaded but not available as a unit file)
         extract_services_from_units
 
+        service_names = services.keys.sort
+        ss = SystemdService.find_many(service_names)
         # Rest of settings
-        services.each_key do |name|
-          services[name][:enabled] = Yast::Service.Enabled(name)
-          services[name][:active] = is_active?(name)
-          if !services[name][:description] || services[name][:description].empty?
-            # Trying to evaluate description via the show command of systemctl
-            s = SystemdService.find(name)
-            services[name][:description] = SystemdService.find(name).description if s
+        service_names.zip(ss).each do |name, s|
+          sh = services[name] # service hash
+          sh[:enabled] = s && s.enabled?
+          sh[:active] = s && s.active?
+          if !sh[:description] || sh[:description].empty?
+            sh[:description] = s.description if s
```

Comment:
so description is nil if service not found? maybe `""` is better?

Reply:
Uh, I don't see a simple way to do this.

Reply:
something like `sh[:description] = s ? s.description : ""` won't work?